### PR TITLE
Add Introspect API for debugging resources across all managed nodes

### PR DESCRIPTION
**Pull request by abhipth**
*Issue #, if available:* #34 

*Description of changes:*
Add introspect API for debugging issues with all Resources across all managed Nodes. This API can later be exposed for customers if required.

Example output.

**Introspect a Node**
```
curl 192.168.6.143:22775/node/ip-192-168-28-99.us-west-2.compute.internal
{
	"vpc.amazonaws.com/PrivateIPv4Address": {
		"UsedResources": {
			"019e0387-df0a-4ec1-a449-76e73a3219c6": "192.168.10.187/19",
			"5a1008ce-7d5e-45fa-8d8c-16b8c9b69b24": "192.168.26.221/19"
		},
		"WarmResources": [
			"192.168.23.206/19",
			"192.168.9.91/19",
			"192.168.30.13/19"
		],
		"CoolingResources": null
	},
	"vpc.amazonaws.com/pod-eni": {}
```

```
curl 192.168.6.143:22775/node/ip-192-168-74-131.us-west-2.compute.internal
{
	"vpc.amazonaws.com/PrivateIPv4Address": {},
	"vpc.amazonaws.com/pod-eni": {
		"TrunkENIID": "eni-0b40ea0edb1313415",
		"InstanceID": "i-03475ffa4ba099884",
		"PodToBranchENI": {
			"7b3fa2e4-e073-41b9-99fd-2f9ed655d0a9": [
				{
					"eniId": "eni-0b5c8ff07fe5e6f7b",
					"ifAddress": "0e:a1:a3:43:70:6b",
					"privateIp": "192.168.68.74",
					"vlanId": 2,
					"subnetCidr": "192.168.64.0/19"
				}
			],
			"81fe2fc0-f805-4e01-876b-a18b44d7472c": [
				{
					"eniId": "eni-0efc320df38a31be0",
					"ifAddress": "0e:09:7b:3b:f3:e5",
					"privateIp": "192.168.75.199",
					"vlanId": 3,
					"subnetCidr": "192.168.64.0/19"
				}
			],
			"92709dc6-796b-4654-9695-9199ecd54ec3": [
				{
					"eniId": "eni-0bb3d659f87398dd2",
					"ifAddress": "0e:53:35:d2:d6:91",
					"privateIp": "192.168.78.19",
					"vlanId": 1,
					"subnetCidr": "192.168.64.0/19"
				}
			]
		},
		"DeleteQueue": null
	}
}
```

**Introspect All Resource**
```
curl 192.168.6.143:22775/resources
{
	"vpc.amazonaws.com/PrivateIPv4Address": {
		"ip-192-168-28-99.us-west-2.compute.internal": {
			"UsedResources": {
				"019e0387-df0a-4ec1-a449-76e73a3219c6": "192.168.10.187/19",
				"5a1008ce-7d5e-45fa-8d8c-16b8c9b69b24": "192.168.26.221/19"
			},
			"WarmResources": [
				"192.168.23.206/19",
				"192.168.9.91/19",
				"192.168.30.13/19"
			],
			"CoolingResources": null
		},
		"ip-192-168-56-131.us-west-2.compute.internal": {
			"UsedResources": {
				"52932b6a-0c8d-4686-bbf1-2259c001a303": "192.168.44.69/19"
			},
			"WarmResources": [
				"192.168.54.216/19",
				"192.168.41.101/19"
			],
			"CoolingResources": null
		},
		"ip-192-168-75-165.us-west-2.compute.internal": {
			"UsedResources": {
				"2d7bad2e-5a26-422f-a241-c09e2c1b83df": "192.168.95.154/19",
				"7c1ab0d5-8e7b-4ae1-8934-af33023d63d0": "192.168.83.72/19",
				"e21d823d-31c6-4b76-9338-7c331dab2b2e": "192.168.79.51/19"
			},
			"WarmResources": [
				"192.168.83.173/19",
				"192.168.86.139/19",
				"192.168.91.236/19"
			],
			"CoolingResources": null
		}
	},
	"vpc.amazonaws.com/pod-eni": {
		"ip-192-168-2-244.us-west-2.compute.internal": {
			"TrunkENIID": "eni-07354395723947518",
			"InstanceID": "i-0e6658551de3e8244",
			"PodToBranchENI": {
				"ac74db91-7e18-42e4-95de-b0eee8999022": [
					{
						"eniId": "eni-04a8dc5d6b0692b2a",
						"ifAddress": "06:7e:9d:a0:d1:6d",
						"privateIp": "192.168.9.211",
						"vlanId": 3,
						"subnetCidr": "192.168.0.0/19"
					}
				]
			},
			"DeleteQueue": null
		},
		"ip-192-168-37-234.us-west-2.compute.internal": {
			"TrunkENIID": "eni-088f9d2edf40f09d5",
			"InstanceID": "i-0bb859268d1666d17",
			"PodToBranchENI": {},
			"DeleteQueue": null
		},
		"ip-192-168-74-131.us-west-2.compute.internal": {
			"TrunkENIID": "eni-0b40ea0edb1313415",
			"InstanceID": "i-03475ffa4ba099884",
			"PodToBranchENI": {
				"7b3fa2e4-e073-41b9-99fd-2f9ed655d0a9": [
					{
						"eniId": "eni-0b5c8ff07fe5e6f7b",
						"ifAddress": "0e:a1:a3:43:70:6b",
						"privateIp": "192.168.68.74",
						"vlanId": 2,
						"subnetCidr": "192.168.64.0/19"
					}
				],
				"81fe2fc0-f805-4e01-876b-a18b44d7472c": [
					{
						"eniId": "eni-0efc320df38a31be0",
						"ifAddress": "0e:09:7b:3b:f3:e5",
						"privateIp": "192.168.75.199",
						"vlanId": 3,
						"subnetCidr": "192.168.64.0/19"
					}
				],
				"92709dc6-796b-4654-9695-9199ecd54ec3": [
					{
						"eniId": "eni-0bb3d659f87398dd2",
						"ifAddress": "0e:53:35:d2:d6:91",
						"privateIp": "192.168.78.19",
						"vlanId": 1,
						"subnetCidr": "192.168.64.0/19"
					}
				]
			},
			"DeleteQueue": null
		}
	}
}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by M00nF1sh — APPROVED** [review]
/lgtm
pretty solid one :D

File: pkg/provider/branch/trunk/trunk_test.go [hunk 1]
```diff
@@ -847,3 +847,20 @@ func TestTrunkENI_CreateAndAssociateBranchENIs_ErrorCreate(t *testing.T) {
...
 	assert.Equal(t, []*ENIDetails{EniDetails1}, trunkENI.deleteQueue)
 }
+
+func TestTrunkENI_Introspect(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	trunkENI, _, mockInstance := getMockHelperInstanceAndTrunkObject(ctrl)
+	trunkENI.trunkENIId = trunkId
+	trunkENI.uidToBranchENIMap[PodUID] = branchENIs1
```
Comment (orsenthil) [inline pkg/provider/branch/trunk/trunk_test.go:857]: A variable naming question. Does `s1` here denote anything in the global var list?

File: pkg/resource/introspect.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	GetAllResourcesPath  = "/resources"
+)
+
+type IntrospectHandler struct {
+	Log             logr.Logger
+	BindAddress     string
+	ResourceManager ResourceManager
+}
+
+// StartENICleaner starts the ENI Cleaner routine that cleans up dangling ENIs created by the controller
```
Comment (orsenthil) [inline pkg/resource/introspect.go:36]: Is this comment on `StartENICleaner` correct here? I couldn't directly see the association.

**Review by orsenthil — COMMENTED** [review]
Excellent feature too. Please see if my comment Start method is correct. Thank you.

File: pkg/resource/introspect.go (hunk 2 above)
Comment (abhipth) [reply pkg/resource/introspect.go:36]: Good Catch, I will fix this in follow up PR!

File: pkg/provider/branch/trunk/trunk_test.go (hunk 1 above)
Comment (abhipth) [reply pkg/provider/branch/trunk/trunk_test.go:857]: It's bad naming convention from my side. It means the first branch ENI ID. I will fix this as well in follow up PR.

**Pull request merged**